File: 프로그래머스/2/42626. 더 맵게/더 맵게.java
import java.util.*;

class Solution {
    public int solution(int[] scoville, int K) {
        int answer = 0;
        
        PriorityQueue<Integer> pq = new PriorityQueue<>();
        
        for(int i=0;i<scoville.length;i++){
            pq.offer(scoville[i]);
        }
        
        while(pq.size()!=1&&pq.peek()<K){
            int food1 = pq.poll();
            int food2 = pq.poll();
            
            int newFood = food1 + (food2*2);
            
            pq.offer(newFood);
            
            answer++;
        }
        
        if(pq.size()==1){
            if(pq.peek()<K) return -1;
        }
        return answer;
    }
}

File: 프로그래머스/2/42626. 더 맵게/더 맵게.cpp
#include <string>
#include <vector>
#include <queue>

using namespace std;

int solution(vector<int> scoville, int K) {
    int answer = 0;
    priority_queue<int,vector<int>,greater<int>> pq;
    
    for(int i=0;i<scoville.size();i++){
        pq.push(scoville[i]);
    }
    
    while(!pq.empty()){
        
        if(pq.top()>=K) break;
        
        if(pq.size()==1){
            if(pq.top()<K){
                answer=-1;
                break;
            }
        }
        
        int food1 = pq.top();
        pq.pop();
        int food2 = pq.top();
        pq.pop();
        
        int newFood = food1+(food2*2);
        pq.push(newFood);
        answer++;
    }
    
    return answer;
}